groestl/src/state.rs

```diff
@@ -151,11 +151,8 @@ impl<BlockSize> GroestlState<BlockSize>
         for i in 0..matrix.rows() {
             for j in 0..matrix.cols() {
                 matrix[i][j] ^= c[i * 16 + j];
-                if c[0] == 0x00 && i == 0 {
-                    matrix[i][j] ^= round;
-                } else if c[0] == 0xff && i == 7 {
-                    matrix[i][j] ^= round;
-                }
```

Comment:
This change is not equivalent and I think the reason behind the test failure.

Reply:
Yes, For some reason I read that as:
```
if a {
    do_something;
} else {
    do_something;
}
```

It should actually be:
```
if a || b {
    do_something;
}
```